lib/src/Base/Optim/openturns/NLopt.hxx

```diff
@@ -38,11 +38,11 @@ class OT_API NLopt :
 public:
 
   /** Default constructor */
-  explicit NLopt(const UnsignedInteger algoType = 0);
+  explicit NLopt(const String algoName = "GN_DIRECT");
```

Comment:
const String & algoName = "LD_SLSQP"
+ a complex type such as String must be passed by reference
+ LD_SLSQP is the most versatile algorithm provided by NLopt